compiler/rustc_resolve/src/late.rs

```diff
@@ -1343,6 +1343,32 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
         }
     }
 
+    fn resolve_defines_attr(&mut self, attrs: &[Attribute]) {
+        for attr in attrs {
+            // for `#[defines(Path, other::Path)]` attributes we need to resolve the paths
+            if attr.has_name(sym::defines) {
+                for item in attr.meta_item_list().unwrap() {
+                    let item = item.meta_item().unwrap();
+                    match self.resolve_path(
+                        &Segment::from_path(&item.path),
+                        Some(Namespace::TypeNS),
+                        None,
```

Comment:
This is not a speculative resolution, so there should be `Some` here, but the real fix would be using `smart_resolve_path` instead of `resolve_path` above.